src/serde_support.rs

```diff
@@ -0,0 +1,266 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+extern crate alloc;
+
+use alloc::format;
+use alloc::string::String;
+
+use dusk_bytes::Serializable;
+use serde::{de, Deserialize, Deserializer, Serialize, Serializer};
+
+use crate::{
+    PublicKey, PublicKeyDouble, PublicKeyVarGen, SecretKey, SecretKeyVarGen,
+    Signature, SignatureDouble, SignatureVarGen,
+};
+
+impl Serialize for PublicKey {
+    fn serialize<S: Serializer>(
+        &self,
+        serializer: S,
+    ) -> Result<S::Ok, S::Error> {
+        let s = bs58::encode(self.to_bytes()).into_string();
+        serializer.serialize_str(&s)
+    }
+}
+
+impl<'de> Deserialize<'de> for PublicKey {
+    fn deserialize<D: Deserializer<'de>>(
+        deserializer: D,
+    ) -> Result<Self, D::Error> {
+        let s = String::deserialize(deserializer)?;
+        let mut bytes = [0; Self::SIZE];
+        let bytes_written = bs58::decode(&s)
+            .into(&mut bytes)
+            .map_err(de::Error::custom)?;
+        if bytes_written == Self::SIZE {
+            Ok(PublicKey::from_bytes(&bytes).map_err(|err| {
+                serde::de::Error::custom(format!("{:?}", err))
+            })?)
```

Comment:
you can use the mapped result directly:

```suggestion
            PublicKey::from_bytes(&bytes).map_err(|err| {
                serde::de::Error::custom(format!("{:?}", err))
            })
```